test/gracenote/search_test.rb

```diff
@@ -2,12 +2,127 @@
+      <TITLE>Come Together</TITLE>
+    </TRACK>
+  </ALBUM>
+  <ALBUM ORD="2">
+    <GN_ID>7317164-1E71D0380B7C84344423FABC8981BC1E</GN_ID>
+    <ARTIST>The Beatles</ARTIST>
+    <TITLE>Sgt. Pepper&apos;s Lonely Hearts Club Band</TITLE>
+    <PKG_LANG>ENG</PKG_LANG>
+    <DATE>1967</DATE>
+    <GENRE NUM="61364" ID="25332">60&apos;s Rock</GENRE>
+    <TRACK_COUNT>13</TRACK_COUNT>
+    <TRACK>
+      <TRACK_NUM>1</TRACK_NUM>
+      <GN_ID>7317165-9AD8800BA4FDB043F7AE8071692CB4BD</GN_ID>
+      <TITLE>Sgt. Pepper&apos;s Lonely Hearts Club Band</TITLE>
+    </TRACK>
+    <TRACK>
+      <TRACK_NUM>2</TRACK_NUM>
+      <GN_ID>7317166-79D0F85136FE2170885F97716EB779F9</GN_ID>
+      <TITLE>With A Little Help From My Friends</TITLE>
+    </TRACK>
+  </ALBUM>
+</RESPONSE>
+</RESPONSES>
+      XML
+    end
+
+    describe "#request" do
+      before do
+        stub_request(:post, //).to_return(headers: { 'Content-Type' => 'applicaiton/xml' }, body: response_xml)
+      end
+
       it "should send mode" do
         client = Gracenote::Client.new(client_id: "test", user_id: "test")
-        allow(client).to receive(:query).with("album_search", hash_including(mode: "SINGLE_BEST_COVER"), a_kind_of(Hash))
+        allow(client).to receive(:post).with(%r{<MODE>SINGLE_BEST_COVER</MODE>}).and_call_original
         client.search(album: "The Bends", mode: "SINGLE_BEST_COVER")
       end
+
+      it "should send auth" do
+        client = Gracenote::Client.new(client_id: "client_id_string", user_id: "user_id_string")
+        allow(client).to receive(:post).with(%r{<AUTH>\s*<CLIENT>client_id_string</CLIENT>\s*<USER>user_id_string</USER>\s*</AUTH>}).and_call_original
+        client.search(album: "The Bends")
+      end
+    end
+
+    it "should return albums" do
+      stub_request(:post, //).to_return(headers: { 'Content-Type' => 'applicaiton/xml' }, body: response_xml)
+
+      client = Gracenote::Client.new(client_id: "test", user_id: "test")
+      response = client.search(artist: "The Beatles")
+      assert { response.ok? == true }
+      assert { response.range.count == 10228 }
+      assert { response.range.start == 1 }
+      assert { response.range.end == 10 }
+      assert { response.range.next == 11 }
+      assert { response.albums.count == 2 }
+      assert { response.albums[0].title == "Abbey Road" }
+      assert { response.albums[0].tracks[0].title == "Come Together" }
+      assert { response.albums[0].tracks[0].track_num == 1 }
+      assert { response.albums[0].genre.name == "60's Rock" }
+      assert { response.albums[0].genre.id == "25332" }
+      assert { response.albums[1].title == "Sgt. Pepper's Lonely Hearts Club Band" }
+      assert { response.albums[1].tracks[1].title == "With A Little Help From My Friends" }
+      assert { response.albums[1].tracks[1].track_num == 2 }
+    end
+
+    it "should return the album info" do
+      stub_request(:post, //).to_return(headers: { 'Content-Type' => 'applicaiton/xml' }, body: <<-XML
```

Comment:
__[rubocop]__
* Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)
* Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)